### PR TITLE
Move Slack source to pkg/ and cmd/

### PR DESCRIPTION
**Pull request by antoineco**
Mostly renaming.

* Relocates `slack/pkg/` to `pkg/` and `slack/cmd` to `cmd/`
* Invokes Slack's `NewAdapter` from shared controller main
* Removes Slack specific release files

Part of #43

### CODE REVIEW

File: config/samples/slacksource.yaml [hunk 1]
```diff
@@ -30,3 +32,24 @@ spec:
...
+
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
```
Comment (antoineco) [inline config/samples/slacksource.yaml:55]: This is a merge of the 3 files previously located at `slack/samples`.

File: pkg/adapter/slacksource/env.go [hunk 2]
```diff
@@ -14,14 +14,14 @@ See the License for the specific language governing permissions and
...
+package slacksource
 
 import (
 	"knative.dev/eventing/pkg/adapter/v2"
 )
 
-// EnvAccesor for configuration parameters
-func EnvAccesor() adapter.EnvConfigAccessor {
+// EnvAccessor for configuration parameters
+func EnvAccessor() adapter.EnvConfigAccessor {
```
Comment (antoineco) [inline pkg/adapter/slacksource/env.go:24]: I used the occasion to fix that typo, I hope to be forgiven.

File: config/samples/slacksource.yaml (hunk 1 above)
Comment (cab105) [reply config/samples/slacksource.yaml:55]: Cool, is this something we'll want to do going forward for our other repos?

Comment (antoineco) [reply config/samples/slacksource.yaml:55]: Maybe. I personally like when samples only provide a rough demonstration of what the _spec_ supports, without falling into the trap of trying to show how the larger ecosystem works. But if we really want to show examples of dependencies in those files, grouping manifests allows for seeing all objects at a glance.

**Pull request merged**